source/Generic/ExtraMetadataToolsVideo/ExtraMetadataToolsVideo.psm1

```diff
@@ -1049,29 +1266,74 @@ function Set-YouTubeVideoManual
     }
 
     $game = $gameDatabase[0]
+    $search = '"' + "https://www.youtube.com/watch?v=" + $YouTubeID + '"'
+
+    $success = Download-YoutubeVideo $game $search $true
+    if($success) 
+    {
+        $PlayniteApi.Dialogs.ShowMessage(([Playnite.SDK.ResourceProvider]::GetString("LOCExtra_Metadata_tools_GenericFinishedMessage")), "Extra Metadata tools")
+    } 
+    else 
+    {
+        $PlayniteApi.Dialogs.ShowMessage(([Playnite.SDK.ResourceProvider]::GetString("LOCExtra_Metadata_tools_GenericErrorProcessingFileMessage")), "Extra Metadata tools")
+    }
+}
+
+function Download-YoutubeVideo
+{
+    param (
+        $game,
+        [string]$search,
+        [bool]$overwrite
+    )
 
     $extraMetadataDirectory = Set-GameDirectory $game
-    $videoPath = Join-Path $extraMetadataDirectory -ChildPath "VideoTrailer.mp4"
-    $videoTempPath = Join-Path $extraMetadataDirectory -ChildPath "VideoTemp.mp4"
-    $youtubedl = $settings.youtubedlPath
-    $search = '"' + "https://www.youtube.com/watch?v=" + $YouTubeID + '"'
-    
+
+    $ytdlp = $settings.ytdlpPath
+    [string]$ytdlpSettings = Get-YTDLPSettings
+
+    $videoPathNoExtension = Join-Path $extraMetadataDirectory -ChildPath "VideoTrailer."
+    $videoTempPathNoExtension = Join-Path $extraMetadataDirectory -ChildPath "VideoTemp."
+
+
+    $filePresent = ((Get-ChildItem -Path "$videoPathNoExtension*").Count -ne 0)
+
+
+    if (($overwrite -eq $false) -and ($filePresent -eq $true))
+    {
+        return $true
+    }
+
+    try {
+        Remove-Item "$videoPathNoExtension*" -Force
+    } catch {}
+
+    try {
+        Remove-Item "$videoTempPathNoExtension*" -Force
+    } catch {}
+
     $trailerdownloadparams = @{
-        'FilePath'     = $youtubedl
-        'ArgumentList' = '-v -o ' + "`"$videoTempPath`"", '-f "mp4"', $search
+        'FilePath'     = $ytdlp
+        'ArgumentList' = @("-o `"$videoTempPathNoExtension%(ext)s`"", $ytdlpSettings, $search)
         'Wait'         = $true
         'PassThru'     = $true
     }
-    $proc = Start-Process @trailerdownloadparams -WindowStyle $settings.youtubedlWindowStyle
+    
+    $proc = Start-Process @trailerdownloadparams -WindowStyle $settings.ytdlpWindowStyle
+
+    $fileTemp = Get-ChildItem -Path "$videoTempPathNoExtension*" | Select BaseName, Extension
```

Comment:
Minor thing, but can you change `Select` alias to the full function name please? `Select-Object`

Reply:
Done in 30b385caf6ed9eef73af37eaed397e5b316ceced